lib/realtime/server.ex

```diff
@@ -0,0 +1,147 @@
+defmodule Realtime.Server do
+  @moduledoc """
+  Fetches live data from RTR, and forwards it to connected clients.
+  """
+
+  use GenServer
+
+  alias Gtfs.Route
+  require Logger
+
+  defmodule Opts do
+    @type t :: %__MODULE__{
+            url: String.t(),
+            poll_delay: integer
+          }
+    @enforce_keys [
+      :url,
+      :poll_delay
+    ]
+    defstruct [
+      :url,
+      :poll_delay
+    ]
+  end
+
+  @type subscriptions :: %{optional(pid()) => [Route.id()]}
+
+  @type vehicles :: %{optional(Route.id()) => [Realtime.Vehicle.t()]}
+
+  @type state :: %{
+          url: String.t(),
+          poll_delay: integer(),
+          subscriptions: subscriptions(),
+          vehicles_timestamp: integer() | nil,
+          vehicles: vehicles()
+        }
+
+  # Client functions
+
+  @spec start_link(Opts.t()) :: GenServer.on_start()
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @spec start_mocked(Opts.t()) :: pid()
+  def start_mocked(opts) do
```

Comment:
Would this be better as an argument to `start_link`? Something like `default_name?: boolean()`?

Reply:
Hmm...

I'd rather have two separate functions for the two different uses, so I don't like another option to `start_link`.

`start_anonymous` isn't quite clear that it's intended for testing. The `start_mocked` name (which `Gtfs` and `HealthServer` also use) came because `Gtfs`, which does mock, was the first.

What about `start_test` or `start_for_testing` or `start_test_server`?

Reply:
Generally I've done the argument to `start_link` version:

```elixir
defmodule Server
  @start_link_opts ~w(name)a

  def start_link(opts) do
    {start_link_opts, opts} = Keyword.split(opts, @start_link_opts)
    GenServer.start_link(__MODULE__, opts, start_link_opts)
  end
end

# supervisor
{Server, name: Server}
```

[ParkingTweets](https://github.com/mbta/parking_tweets/blob/master/lib/parking_tweets/application.ex#L22-L25) has an example too.